src/test/java/com/web/parallel/BaseTest.java

```diff
@@ -0,0 +1,106 @@
+        DesiredCapabilities caps = new DesiredCapabilities();
+        caps.setCapability("project", "BrowserStack Random Browsers");
+        caps.setCapability("browserstack.user", USERNAME);
+        caps.setCapability("browserstack.key", ACCESS_KEY);
+        List<BrowserDetails> browsers = get("browsers.json")
+                .jsonPath()
+                .getList("", BrowserDetails.class);
+        int randomNumber;
+        BrowserDetails browserDetails;
+        switch (environment) {
+            case "desktop":
+                List<BrowserDetails> desktopBrowsers = browsers.parallelStream()
+                        .filter(browser -> browser.getDevice() == null)
+                        .collect(toList());
+                randomNumber = ThreadLocalRandom.current().nextInt(0, desktopBrowsers.size());
+                browserDetails = desktopBrowsers.get(randomNumber);
+                System.out.println(browserDetails);
+                caps.setCapability("build", "Random Desktop Browsers - " + TIMESTAMP);
+                caps.setCapability("name", m.getName() + " - " + browserDetails.getBrowser() + " " + browserDetails.getBrowser_version());
+                caps.setCapability("os", browserDetails.getOs());
+                caps.setCapability("os_version", browserDetails.getOs_version());
+                caps.setCapability("browser", browserDetails.getBrowser());
+                caps.setCapability("browser_version", browserDetails.getBrowser_version());
+                break;
+            case "mobile":
+                List<BrowserDetails> mobileBrowsers = browsers.parallelStream()
+                        .filter(BrowserDetails::isReal_mobile)
+                        .collect(toList());
+                randomNumber = ThreadLocalRandom.current().nextInt(0, mobileBrowsers.size());
+                browserDetails = mobileBrowsers.get(randomNumber);
+                System.out.println(browserDetails);
+                caps.setCapability("build", "Random Mobile Browsers - " + TIMESTAMP);
+                caps.setCapability("name", m.getName() + " - " + browserDetails.getDevice());
+                caps.setCapability("os_version", browserDetails.getOs_version());
+                caps.setCapability("device", browserDetails.getDevice());
+                caps.setCapability("browser", browserDetails.getBrowser());
+                caps.setCapability("real_mobile", browserDetails.isReal_mobile());
+                break;
+            default:
+                throw new IllegalArgumentException("Incorrect environment " + environment + " specified");
+        }
+        driverThread.set(new RemoteWebDriver(new URL(URL), caps));
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void teardown() {
+        JavascriptExecutor js = (JavascriptExecutor) driverThread.get();
+        js.executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \"passed\"}}");
```

Comment:
All tests are passed even when elements are not found. Need to implement correct test pass/fail logic

Reply:
I'm just using this to mark tests as passed as the feature that this code is showcasing does not require to check whether a particular test passed or failed.